packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1534,10 +1548,20 @@ export class ImportResolver {
             }
 
             // Prefer local packages.
-            if (bestImportSoFar.importType === ImportType.Local && !bestImportSoFar.isNamespacePackage) {
+            if (
+                bestImportSoFar.importType === ImportType.Local &&
+                !bestImportSoFar.isNamespacePackage &&
+                newImport.importType !== ImportType.Local
+            ) {
                 return bestImportSoFar;
             }
-            if (newImport.importType === ImportType.Local && !newImport.isNamespacePackage) {
+
+            // Prefer higher priority.
+            if (
+                newImport.importType === ImportType.Local &&
+                !newImport.isNamespacePackage &&
+                newImport.priority > bestImportSoFar.priority
```

Comment:
I don't think this is correct. The import resolution ordering shouldn't be affected by the import type or priority. It's fine to change the completion ordering based on the priority, but the import resolution ordering is dictated by the way the Python import loader works.

Reply:
The user is asking to control just that. Perhaps there is another way?
https://github.com/microsoft/pylance-release/issues/3429

Like maybe we support .env files?

Reply:
I think there's some confusion here about the meaning of "exclude". It indicates that the type checker shouldn't run on those files. It has no bearing on import resolution. If the user is asking to control import resolution using "exclude", then you should correct their misunderstanding.

Reply:
They understand what is used for, but they want something new. This was one way to accomplish their goal, I'm wondering if there are others. 

Extra paths is used for controlling import resolution, perhaps it should behave like insert instead of append or have some other way to insert

Reply:
It wouldn't be appropriate to overload the use of "exclude" for this purpose. These are very different concepts.

Is the use case important enough to bother with? It was reported by one user and has no upvotes. Maybe just say "won't fix" or open for discussion.

Reply:
Fair point. Ill close this then